go/vt/vtgate/executor.go

```diff
@@ -683,6 +683,14 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 	defer func() { logStats.ExecuteTime = time.Since(execStart) }()
 
 	switch show.Type {
+	case sqlparser.KeywordString(sqlparser.VARIABLES):
+		if destKeyspace == "" {
+			keyspaces, err := e.resolver.resolver.GetAllKeyspaces(ctx)
+			if err !=nil && len(keyspaces) == 0 {
```

Comment:
Shouldn't this be an `||` condition? Also, if len(keyspaces)==0, err could be nil. So, you'll need to generate your own error message.

Reply:
Sorry my fault, just pushed without thinking.
Pushing now the good one.